source/robot_model/src/Model.cpp

```diff
@@ -184,6 +184,30 @@ bool Model::check_collision(const state_representation::JointPositions& joint_po
   return false;
 }
 
+// Compute minimum distance
+Eigen::MatrixXd Model::compute_minimum_distance(const state_representation::JointPositions& joint_positions) {
+    Eigen::VectorXd configuration = joint_positions.get_positions();
+    pinocchio::computeDistances(this->robot_model_, this->robot_data_, this->geom_model_, this->geom_data_, configuration);
+    
+    // nb_joints is the number of joints in the robot model
+    unsigned int nb_joints = this->get_number_of_joints();
+    
+    // create a 2D Eigen vector to store the distances and initialize to zero
+    Eigen::MatrixXd distances = Eigen::MatrixXd::Zero(nb_joints, nb_joints);
+    
+    // iterate over the collision pairs and extract the distances
+    unsigned int pair_index = 0;
+    for (unsigned int row_index = 0; row_index < nb_joints; ++row_index) {
+        for (unsigned int column_index = row_index+1; column_index < nb_joints; ++column_index) {
+          distances(row_index, column_index) = this->geom_data_.distanceResults[pair_index].min_distance;
+          distances(column_index, row_index) = distances(row_index, column_index);
```

Comment:
You're essentially filling a matrix with duplicated data, could we not say that the resulting matrix is only upper or lower triangular?

Reply:
Yes we can indeed! Initially I had in mind to output a symmetric distance matrix so that any user would find intuitive to find a distance between link i & j then it is d(i, j) or d(j ,i). We can on the other hand, make it clear in the documentation.

Reply:
I think the most important part is breaking out of the calculation early (avoiding iteration of the whole square matrix). This is already done with the `column_index < nb_joints`. Writing the data to the other side of the matrix is not a huge burden IMO (because these matrices will generally not be massive in size) and it improves user experience. So, I don't mind it how it is. But I appreciate the suggestion and attention to detail!